File: Taller Nº5/python/NumeroPerfecto.py

def es_numero_perfecto(n): # Declara una función que recibe un número entero n y retorna si es perfecto (True o False).
    return n > 1 and sum(i for i in range(1, n) if n % i == 0) == n

def main():
    numero = int(input("Ingresa un número: "))
    print(f"{numero} es un número perfecto." #" si es_numero_perfecto(numero)
        # else f"{numero} no es un número perfecto.")
        if es_numero_perfecto(numero) # La función retorna True solo si ambas condiciones se cumplen
        else f"{numero} no es un número perfecto.") # La función retorna Falsa si alguna de las condiciones no se cumple

main()

File: Taller Nº5/Java/NumeroPerfecto.java

import java.util.Scanner;
import java.util.stream.IntStream;

public class Main {

    // Función que retorna un mensaje como String, en lugar de booleano
    public static String verificarNumeroPerfecto(int n) {
        int suma = IntStream.range(1, n)
                .filter(i -> n % i == 0)
                .sum();

        return (n > 1 && suma == n) ?
                "Es un número perfecto." :
                "No es un número perfecto.";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingresa un número: ");
        int numero = sc.nextInt();

        // Mostrar resultado directamente
        System.out.println(numero + ": " + verificarNumeroPerfecto(numero));
    }
}